pkg/apis/ceph.rook.io/v1/types.go

```diff
@@ -1102,6 +1102,12 @@ type GatewaySpec struct {
 	// +nullable
 	// +optional
 	ExternalRgwEndpoints []v1.EndpointAddress `json:"externalRgwEndpoints,omitempty"`
+
+	// The annotations-related configuration to add/set on each rgw service.
+	// +kubebuilder:pruning:PreserveUnknownFields
```

Comment:
I don't believe this is necessary for `map[string]` types. Please check this; it will allow us to generate smaller CRDs.

Reply:
For passing into the `ServiceObjectMeta` this need to `rookv1.Annotations`. In future, we may need more annotation for the rgw service.

Reply:
Without the preserUnknownFields, are the annotations not preserved? From the schema generation seems like it will be fine without that property.